addons/mail/static/src/components/message/message.js

```diff
@@ -433,20 +349,18 @@ class Message extends Component {
             return;
         }
         if (this._prettyBodyRef.el && this.message.prettyBody !== this._lastPrettyBody) {
-            this._prettyBodyRef.el.innerHTML = this.message.prettyBody;
-            this._lastPrettyBody = this.message.prettyBody;
-        }
-        // Remove all readmore before if any before reinsert them with _insertReadMoreLess.
-        // This is needed because _insertReadMoreLess is working with direct DOM mutations
-        // which are not sync with Owl.
-        if (this._contentRef.el) {
-            for (const el of [...this._contentRef.el.querySelectorAll(':scope .o_Message_readMoreLess')]) {
-                el.remove();
+            this._prettyBodyRef.el.innerHTML = '';
+            if (
+                this.message.prettyBodyWithReadMoreLess &&
+                this.message.prettyBodyWithReadMoreLess.children().length > 0
```

Comment:
When does it have children or not, and why condition it?
If it didn't have children for some reason, I would be expecting it to display whatever is "no children", instead of falling back on `prettyBody`.

Reply:
This is to support cases with `prettyBody` being pure textNode.